### PR TITLE
feat: Create Warehouse Functionality

### LINKED ISSUE #5: Create Warehouse Functionality
### Create Warehouse Functionality

Create, update, and delete warehouses.

Subtasks:

- [x] Implement create warehouse API.
- [x] Implement update warehouse API.
- [x] Implement delete warehouse API.

### PR DESCRIPTION
**Pull request by lmv2004**

### CODE REVIEW

**Comment by lmv2004** [comment]
anh ơi review giúp em với

File: server/src/main/java/com/warehousepro/controller/WareHouseController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@FieldDefaults(level = AccessLevel.PRIVATE , makeFinal = true)
+@RequestMapping("/warehouses")
+public class WareHouseController {
+
+    WareHouseService warehouseService;
+
+    WareHouseMapper warehouseMapper;
+
+    @GetMapping
+    public List<WareHouseResponseDto> getAllWarehouse() {
```
Comment (tnowad) [inline server/src/main/java/com/warehousepro/controller/WareHouseController.java]: It’s not ideal to return a list of `WarehouseResponseDto` directly. Instead, consider creating a paginated response DTO as outlined in this example:  
[Pagination Example - REST API Design Guide](https://github.com/NationalBankBelgium/REST-API-Design-Guide/wiki/Pagination-Example).

This approach improves scalability and ensures consistency with API best practices.

File: server/src/main/java/com/warehousepro/controller/WareHouseController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return warehouseMapper.toWareHouseResponse(warehouseService.createWareHouse(warehouseRequest));
+    }
+
+    @PutMapping("/{id}")
+    public WareHouseResponseDto updateWarehouse(@PathVariable("id") int id, @RequestBody WareHouseUpdateRequestDto warehouseRequest) {
+        return warehouseMapper.toWareHouseResponse(warehouseService.updateWareHouse(id, warehouseRequest));
+    }
+
+    @DeleteMapping("/{id}")
+    public String deleteWarehouse(@PathVariable("id") int id) {
```
Comment (tnowad) [inline server/src/main/java/com/warehousepro/controller/WareHouseController.java]: > 204 No Content - Response to a successful request that won't be returning a body (like a DELETE request)

File: server/src/main/java/com/warehousepro/service/WareHouseService.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@Service
+@RequiredArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE , makeFinal = true)
+public class WareHouseService {
+
+    WareHouseRepository warehouseRepository;
+
+    WareHouseMapper warehouseMapper;
+
+    public List<Warehouse> getAllWareHouse() {
```
Comment (tnowad) [inline server/src/main/java/com/warehousepro/service/WareHouseService.java]: It should support pagination, sort, and filter

File: server/src/main/java/com/warehousepro/service/WareHouseService.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        return warehouseRepository.findAll(); //not supported Exception
+    }
+
+    public Warehouse getWareHouse(int warehouse_id) {
```
Comment (tnowad) [inline server/src/main/java/com/warehousepro/service/WareHouseService.java]: Please follow the java coding rule, don't use `snake_case` in here

**Review by tnowad — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/warehousepro/service/WareHouseService.java (hunk 3 above)
Comment (tnowad) [reply server/src/main/java/com/warehousepro/service/WareHouseService.java]: https://docs.spring.io/spring-data/jpa/reference/jpa/specifications.html

https://github.com/xm-online/xm-ms-entity/blob/65b05f25db775fb2582710b6a856226d697279cb/src/main/java/com/icthh/xm/ms/entity/service/query/EventQueryService.java#L49

**Review by tnowad — APPROVED** [review]

**Pull request merged**